pkg/sql/drop_sequence.go

```diff
@@ -190,7 +189,7 @@ func (p *planner) canRemoveOwnedSequencesImpl(
 		var firstDep *descpb.TableDescriptor_Reference
 		multipleIterationErr := seqDesc.ForeachDependedOnBy(func(dep *descpb.TableDescriptor_Reference) error {
 			if firstDep != nil {
-				return iterutil.StopIteration()
+				return errors.Newf("multiple iterations")
```

Comment:
If StopIteration is returned then the error will be nil.

This also fixes an error which can be tested with
```

statement ok
CREATE TABLE test1 (a int GENERATED ALWAYS AS IDENTITY (START WITH 10));

statement ok
CREATE TABLE test2(id INT PRIMARY KEY DEFAULT nextval('public.test1_a_seq'));

statement ok
SET use_declarative_schema_changer = off

# this should error
statement ok
drop table test1;

statement error descriptor is being dropped
INSERT INTO test2 DEFAULT VALUES;
```